utils/motorcontroller/SparkMAXFactory.java

```diff
@@ -75,6 +71,7 @@ public static SparkMAXLite buildSparkMAX(int deviceID, Config config, MotorConfi
         sparkMax.setInverted(config.INVERTED);
         sparkMax.setSmartCurrentLimit(MotorConfig.MOTOR_MAX_CURRENT.get(motorType));
         sparkMax.enableVoltageCompensation(12);
+        // TODO set status frames
```

Comment:
I think this is a good start. The trick is going to be that if you look at our 2022 code, this function is only called in one place. The others are calling different functions that are picking the default config (e.g. buildFactorySparkMAX()). This where you're an API designer and have to think about tradeoffs. One answer could be that anywhere that we care about it we should stop doing that and call this one instead. And maybe that's the right answer. Another answer could be to get rid of the buildFactorySparkMAX() and force the caller to think about what kind of control they're asking for. That could be by providing some alternate convenience functions (e.g. buildVelocityControlledSparkMax(), etc.) or it could be by just requiring them to call this function with the appropriate configuration. Or there may be other options. I think it's worth a little thought and probably some conversation

Reply:
I agree with Mr. Dias on needing alternate convenience functions because most people on the software team (including me) generally just use buildSparkMax without think any time we need a motor object